src/app/chat/chat-routing.module.ts

```diff
@@ -7,10 +7,11 @@ import { ChatRoomComponent } from './chat-room/chat-room.component';
 const routes: Routes = [
   {
     path: '',
-    component: ChatComponent
+    redirectTo: '/chat/chat-room',
+    pathMatch: 'full'
   },
   {
-    path: 'chatroom',
+    path: 'chat-room',
```

Comment:
if we include id that means we can call api using that id and get selected chat data or we can search it on our cashed data. To get one chat object no need to make api call, and also we didn't cash chat data because it's a real time function. We can pass selected chat data from chat component to chat room component. and also we no need to routes for team and one to one. because in api call for the get messages only **team_member_id** is the one not sending to team message, **team_id** is always sending. So if we pass selected chat object we can check chat type from it and modify api call in chat room component.